packages/flutter/lib/src/material/flexible_space_bar.dart

```diff
@@ -262,9 +262,9 @@ class FlexibleSpaceBarSettings extends InheritedWidget {
   /// [FlexibleSpaceBar] widget in its tree for the settings to take affect.
   const FlexibleSpaceBarSettings({
     Key key,
-    this.toolbarOpacity,
-    this.minExtent,
-    this.maxExtent,
+    @required this.toolbarOpacity,
+    @required this.minExtent,
+    @required this.maxExtent,
```

Comment:
Should assert that these parameters are not null, that minExtent < maxExtent, that toolbarOpacity >=  0

Reply:
Can I assert minExtent <= maxExtent? If I assert minExtent < maxExtent most of the tests fail

Reply:
Sorry about that, yes: minExtent <= maxExtent is correct.

Reply:
Added these, is it also worth asserting minExtent > 0  and maxExtent > 0?

Reply:
[sorry, I was out for a week] Yes, asserting that both extents are > 0 before checking them against currentExtent sounds like a good idea.